fem/tmop_tools.cpp

```diff
@@ -218,6 +218,77 @@ void ParAdvectorCGOper::Mult(const Vector &ind, Vector &di_dt) const
 }
 #endif
 
+#ifdef MFEM_USE_GSLIB
+void InterpolatorFP::SetInitialField(const Vector &init_nodes,
+                                     const Vector &init_field)
+{
+   const bool serial = !parallel;
+   nodes0 = init_nodes;
+   field0 = init_field;
+   Mesh *m = mesh;
+#ifdef MFEM_USE_MPI
+   if (pmesh) { m = pmesh; }
+#endif
+   m->SetNodes(nodes0);
+
+   const double rel_bbox_el = 0.1;
+   const double newton_tol  = 1.0e-12;
+   const int npts_at_once   = 256;
+
+#ifdef MFEM_USE_MPI
+   if (!serial) {finder = new FindPointsGSLIB(pfes->GetComm());}
+   else         {finder = new FindPointsGSLIB();}
+#else
+   finder = new FindPointsGSLIB();
+#endif
+   finder->Setup(*m, rel_bbox_el, newton_tol, npts_at_once);
+}
+
+void InterpolatorFP::ComputeAtNewPosition(const Vector &new_nodes,
+                                          Vector &new_field)
+{
+
+   FiniteElementSpace *f = fes;
+#ifdef MFEM_USE_MPI
+   if (pfes)  { f = pfes; }
+#endif
+
+   const int dim = f->GetFE(0)->GetDim();
+   Vector positionsToFind = new_nodes;
+   const int pts_cnt = positionsToFind.Size() / dim;
+
+
+   // Interpolate FE function values on the found points.
+   Array<uint> el_id_out(pts_cnt), code_out(pts_cnt), task_id_out(pts_cnt);
+   Vector pos_r_out(pts_cnt*dim), dist_p_out(pts_cnt);
```

Comment:
These are the same size every time, it might be better to move them in the class instead of allocating at each call.

Reply:
@vladotomov how about we move these inside the gslib class so that the interface to FindPoints and Interpolate is much simpler:
` FindPoints(Vector &point_pos) `
`Interpolate(const GridFunction &field_in, Vector &field_out) `

Reply:
Yes that would actually be better. Then we can have Get methods if the user actually needs the extra info.

However that would change the public interface (I know of one project that uses the current interface). We can have another PR and discuss there.